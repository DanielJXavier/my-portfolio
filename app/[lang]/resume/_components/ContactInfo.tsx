import { useParams } from "next/navigation";

import { getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import { author } from "@/_config";

import { links } from "@/[lang]/(pages)/(home)/_config";

import VisibilityControl from "./VisibiltyControl";

const commonClassName =
  "print:relative print:flex print:items-center print:after:content-[''] print:after:absolute print:after:-right-2 print:after:w-px print:after:h-3 print:after:bg-primary";

export default function ContactInfo() {
  const { lang } = useParams<{ lang: Lang }>();

  const {
    resume: { locationInfo },
  } = getDictionary(lang);

  return (
    <section>
      <h1 className="text-[18pt] font-bold text-center">
        {author.firstName} {author.lastName}
      </h1>

      <p className="text-center">
        {locationInfo.city}, {locationInfo.state}, {locationInfo.country}
      </p>

      <ul className="print:flex print:gap-x-4 print:justify-center">
        <VisibilityControl className={commonClassName}>
          {links.email}
        </VisibilityControl>

        <VisibilityControl className={commonClassName}>
          {links.whatsApp}
        </VisibilityControl>

        <VisibilityControl className={commonClassName}>
          <a
            className="underline"
            href={`https://www.linkedin.com/in/${links.linkedIn}/`}
            target="_blank"
          >
            linkedin.com/in/{links.linkedIn}
          </a>
        </VisibilityControl>

        <VisibilityControl>
          <a
            className="underline"
            href={`https://www.${links.site}/`}
            target="_blank"
          >
            {links.site}
          </a>
        </VisibilityControl>
      </ul>
    </section>
  );
}
