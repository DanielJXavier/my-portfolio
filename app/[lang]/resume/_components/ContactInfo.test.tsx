import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import ContactInfo from "./ContactInfo";

import { useParams } from "next/navigation";

jest.mock("next/navigation");

(useParams as jest.Mock).mockImplementation(() => ({ lang: "en" }));

describe("ContactInfo component", () => {
  it("renders the component", () => {
    const { container } = render(<ContactInfo />);

    expect(container).toMatchSnapshot();
  });
});
