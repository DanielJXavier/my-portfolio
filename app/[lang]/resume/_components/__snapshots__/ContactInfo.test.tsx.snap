// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfo component renders the component 1`] = `
<div>
  <section>
    <h1
      class="text-[18pt] font-bold text-center"
    >
      Daniel
       
      Xavier
    </h1>
    <p
      class="text-center"
    >
      Florian√≥polis
      , 
      Santa Catarina
      , 
      Brazil
    </p>
    <ul
      class="print:flex print:gap-x-4 print:justify-center"
    >
      <li
        class="print:relative print:flex print:items-center print:after:content-[''] print:after:absolute print:after:-right-2 print:after:w-px print:after:h-3 print:after:bg-primary flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        dxdanielxavier@gmail.com
      </li>
      <li
        class="print:relative print:flex print:items-center print:after:content-[''] print:after:absolute print:after:-right-2 print:after:w-px print:after:h-3 print:after:bg-primary flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        +55 31 99732-5804
      </li>
      <li
        class="print:relative print:flex print:items-center print:after:content-[''] print:after:absolute print:after:-right-2 print:after:w-px print:after:h-3 print:after:bg-primary flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <a
          class="underline"
          href="https://www.linkedin.com/in/danieljxavier/"
          target="_blank"
        >
          linkedin.com/in/
          danieljxavier
        </a>
      </li>
      <li
        class=" flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <a
          class="underline"
          href="https://www.danielxavier.dev/"
          target="_blank"
        >
          danielxavier.dev
        </a>
      </li>
    </ul>
  </section>
</div>
`;
